storage/interface.go

```diff
@@ -294,6 +295,14 @@ type MetadataUpdater interface {
 	UpdateMetadata(ref SeriesRef, l labels.Labels, m metadata.Metadata) (SeriesRef, error)
 }
 
+// CreatedTimestampAppender provides an interface for appending created timestamps to the storage.
+type CreatedTimestampAppender interface {
+	// AppendCreatedTimestamp adds an extra sample to the given series labels.
+	// The value of the appended sample is always zero, while the sample's timestamp
+	// is the one exposed by the target as created timestamp.
+	AppendCreatedTimestamp(ref SeriesRef, l labels.Labels, t int64) (SeriesRef, error)
```

Comment:
Yeah makes sense. How do you usually communicate this kind of situation? Are comments enough?

Reply:
> Should AppendCreatedTimestamp error out if they can't append CT (e.g. because new sample exists older than CT (OOO) and we already ingested (or not) zero) or should return no error?

That's a good question... I believe we should be able to send debug logs when created timestamps are ignored, but we won't have a logger inside the AppendCreatedTimestamp method. Should we also return some sort of warning that could be logged by whatever calls AppendCreatedTimestamp? WDYT?

Reply:
Actually, I went with returning a typed error when out of order, the caller can identify and log if needed

Reply:
Comment is ok, but whatever we do let's be consistent and explicit instead of giving next code author opportunity to get surprised (: 

> Actually, I went with returning a typed error when out of order, the caller can identify and log if needed

I like that.